src/coord/src/coord/dataflow_builder.rs

```diff
@@ -112,6 +112,9 @@ impl<'a> DataflowBuilder<'a> {
 
             // A valid index is any index on `id` that is known to the dataflow
             // layer, as indicated by its presence in `self.compute`.
+            //
+            // TODO: indexes should be imported after the optimization process,
+            // and only those actually used by the optimized plan
```

Comment:
LIR generation has index pruning as part of it, if this is at all reassuring. As it forms its plan it binds more strongly to the available indexes, and can then see which ones are not used. I think it deletes them from `index_imports`.

Reply:
That is very reassuring!